TestResultSummaryService/parsers/BenchmarkParser.js

```diff
@@ -89,18 +88,34 @@ class BenchmarkParser extends Parser {
         const { javaVersion, jdkDate } = this.exactJavaVersion( output );
         const { nodeVersion, nodeRunDate} = this.exactNodeVersion( output );
 
-        return {
-            tests,
-            jdkDate,
-            javaVersion,
-            nodeRunDate,
-            nodeVersion,
-            buildResult,
-            machine: this.extractMachineInfo( output ),
-            type: "Perf",
-            startBy: this.extractStartedBy( output ),
-            artifactory: this.extractArtifact( output ),
-        };
+        if (javaVersion && jdkDate) {
+            return {
+                tests,
+                sdkResource: productResource,
+                jdkDate,
+                javaVersion,
+                buildResult,
+                machine: this.extractMachineInfo( output ),
+                type: "Perf",
+                startBy: this.extractStartedBy( output ),
+                artifactory: this.extractArtifact( output ),
+            };
+        } else if (nodeRunDate && nodeVersion){
+            return {
+                tests,
+                sdkResource: productResource,
+                nodeRunDate,
+                nodeVersion,
+                buildResult,
+                machine: this.extractMachineInfo( output ),
+                type: "Perf",
+                startBy: this.extractStartedBy( output ),
+                artifactory: this.extractArtifact( output ),
+            };
+        } else {
+            return;
+        }
```

Comment:
The above if-else blocks look weird to me. If version and date are required, then we should not do the loop at all. Fail when the version or date is invalid. I think if we want to fix it, we need to fix it properly - have a general version and date. We have an issue opened for this https://github.com/AdoptOpenJDK/openjdk-test-tools/issues/167

Reply:
@piyush286, may I have more suggestions here.

Reply:
@sophiaxu0424 In order to focus on clean up issues and critical fixes, let's not fix the java and node issue (#167).  Hence, we can leave this if-else block change for now since I feel the same as Lan. We can just move `sdkResource` to top level as you've done and add it to the original JSON data that's being returned.